src/Foundatio.AzureServiceBus/Messaging/AzureServiceBusMessageBus.cs

```diff
@@ -42,36 +45,72 @@ public class AzureServiceBusMessageBus : MessageBusBase<AzureServiceBusMessageBu
 
                 var sw = Stopwatch.StartNew();
                 try {
-                    await _namespaceManager.CreateSubscriptionAsync(CreateSubscriptionDescription()).AnyContext();
-                } catch (MessagingEntityAlreadyExistsException) { }
+                    var sbManagementClient = await GetManagementClient().AnyContext();
+                    if (sbManagementClient != null) {
+                        await sbManagementClient.Subscriptions.CreateOrUpdateAsync(_options.ResourceGroupName, _options.NameSpaceName,
+                            _options.Topic, _subscriptionName, CreateSubscriptionDescription()).AnyContext();
+                    }
+                }
+                catch (ErrorResponseException) { }
 
                 // Look into message factory with multiple recievers so more than one connection is made and managed....
-                _subscriptionClient = SubscriptionClient.CreateFromConnectionString(_options.ConnectionString, _options.Topic, _subscriptionName, ReceiveMode.ReceiveAndDelete);
-                _subscriptionClient.OnMessageAsync(OnMessageAsync, new OnMessageOptions { /* AutoComplete = true, // Don't run with recieve and delete */ MaxConcurrentCalls = 6 /* calculate this based on the the thread count. */ });
-                if (_options.SubscriptionRetryPolicy != null)
-                    _subscriptionClient.RetryPolicy = _options.SubscriptionRetryPolicy;
+                _subscriptionClient = new SubscriptionClient(_options.ConnectionString, _options.Topic, _subscriptionName, _options.ReceiveMode, _options.SubscriptionRetryPolicy);
+                
+                // See if this blows up?
+                
+                // Enable prefetch to speeden up the receive rate.
                 if (_options.PrefetchCount.HasValue)
                     _subscriptionClient.PrefetchCount = _options.PrefetchCount.Value;
 
+                // these are the default values of MessageHandlerOptions
+                int maxConcurrentCalls = 1;
+                bool autoComplete = true;
+                if (_options.MaxConcurrentCalls.HasValue)
+                    maxConcurrentCalls = _options.MaxConcurrentCalls.Value;
+
+                if (_options.AutoComplete.HasValue)
+                    autoComplete = _options.AutoComplete.Value;
+
+                _subscriptionClient.RegisterMessageHandler(OnMessageAsync, new MessageHandlerOptions(OnExceptionAsync) { MaxConcurrentCalls = maxConcurrentCalls, AutoComplete = autoComplete });
                 sw.Stop();
                 _logger.Trace("Ensure topic subscription exists took {0}ms.", sw.ElapsedMilliseconds);
             }
         }
 
-        private async Task OnMessageAsync(BrokeredMessage brokeredMessage) {
+        // Use this Handler to look at the exceptions received on the MessagePump
+        private Task OnExceptionAsync(ExceptionReceivedEventArgs args) {
+            _logger.Warn(args.Exception, "Message handler encountered an exception.");
```

Comment:
Do not swallow `args`. It contains valuable information valuable information such as 
- action that failed
- entity path
- namespace
It all should be logged imo

Reply:
+1, We upgraded to the new Microsoft logging in foundatio 6. I can pair on this if you'd like